src/BlockEntities/FurnaceEntity.cpp

```diff
@@ -169,6 +169,25 @@ bool cFurnaceEntity::ContinueCooking(void)
 	return m_IsCooking;
 }
 
+int cFurnaceEntity::GetReward(void)
+{
+	float RawReward = m_CurrentRecipe->Reward;
```

Comment:
Doesn't `m_CurrentRecipe` get reset if the furnace finishes cooking completely (input slot gets empty)?
At the very least this function should check if `m_CurrentRecipe` is valid, and short-circuit return zero when not.